### ISSUE TITLE
Missing bottom borders in Orders page with Gift Messages allowed

### ISSUE THREAD

**Issue by ADDISON74** [OpenMage/magento-lts#2200]
Go to Backend > System > Sales. In "Gift Options" section set as follows:

- Allow Gift Message on Order = Yes
- Allow Gift Message for Order = Yes

Create an order in Backend and place it. Take a look to "Items Ordered" grid. Both rows do not have bottom borders. 

![borders](https://user-images.githubusercontent.com/8360474/172254123-a2bb7e7b-abc5-48e8-963d-5510324ad910.jpg)

Let's inspect these rows. As you may see with Gifts allowed <tr> tag doesn't have any class. 

![inspector-1](https://user-images.githubusercontent.com/8360474/172254982-25400aa6-16ac-428e-a619-76eda2f13817.jpg)

When "Allow Gifts Message on/for Order" are set to No, both rows has ```class="border"```. Just create a new order after you set the values to No and see what is happening.

Let's add this class in browser Inspector for the first row and look what happened. It got border bottom.

![inspector-2](https://user-images.githubusercontent.com/8360474/172255284-9a3c2774-a928-4e77-98f4-abb570eb7127.jpg)

In conclusion, when "Allow Gifts" is set to Yes there is a template used in this case but ```class="border"``` is missing from it. I did not have time to identify the template but I will do it soon. It took me some time to figure out what is going on, initially I was convinced it was a missing template or CSS issue but it wasn't.

**Comment by ADDISON74** [comment]
The template file responsible for this issue I found it here

/app/design/adminhtml/default/default/template/sales/order/view/items/renderer/default.phtml

From line 27:

```php
<?php $_item = $this->getItem() ?>
<?php $this->setPriceDataObject($_item) ?>
<tr<?php if (!$this->canDisplayGiftmessage()): ?> class="border"<?php endif ?>>
    <td>
        <?php if ($this->canDisplayContainer()): ?>
        <div id="<?php echo $this->getHtmlId() ?>" class="item-container">
        <?php endif ?>
        <div class="item-text">
            <?php echo $this->getColumnHtml($_item, 'name') ?>
        </div>
        <?php if ($this->canDisplayContainer()): ?>
        </div>
        <?php endif ?>
    </td>
```

As I anticipated border class is not added when Allow Gift Message is set to Yes.

```php
<tr<?php if (!$this->canDisplayGiftmessage()): ?> class="border"<?php endif ?>>
```

After removing ! in if statement here is the "Items Ordered grid, both rows have border bottom:

![border](https://user-images.githubusercontent.com/8360474/172324587-d1251683-4bd2-4054-9c79-365110522aec.jpg)

Let's go further with the investigation. This file

/app/design/adminhtml/default/default/template/giftmessage/sales/order/view/items.phtml

is responsible for adding in every row the "Gift Options" link. I added ! in line 28 ```<?php if (!$this->canDisplayGiftmessage()): ?> ``` and I got the link for every item without creating a new order.

![gift-options](https://user-images.githubusercontent.com/8360474/172326055-558968a3-1870-4eef-afcb-eded89a15c13.jpg)

As you may see the border class I added before for the situation when Gift Messages are allowed did not create any issues.

In conclusion the bug is confirmed. What is not clear to me is why the author of the code ruled out adding the border class in this particular situation. I guess the idea that he had a reason to introduce that if statement into the code.
